src/main/java/in/org/projecteka/hiu/HiuConfiguration.java

```diff
@@ -430,16 +434,32 @@ public JWKSet jwkSet(GatewayProperties gatewayProperties) throws IOException, Pa
         return JWKSet.load(new URL(gatewayProperties.getJwkUrl()));
     }
 
+    @Bean("identityServiceJWKSet")
+    public JWKSet identityServiceJWKSet(IdentityServiceProperties identityServiceProperties) throws IOException, ParseException {
+        return JWKSet.load(new URL(identityServiceProperties.getJwkUrl()));
+    }
+
     @Bean
     public GatewayTokenVerifier centralRegistryTokenVerifier(@Qualifier("centralRegistryJWKSet") JWKSet jwkSet) {
         return new GatewayTokenVerifier(jwkSet);
     }
 
-    @Bean
+    @Bean("hiuUserAuthenticator")
```

Comment:
call it admins authenticator? I know doctors also use it.

Reply:
but this instance I don't think doctor role will have access to it.

Reply:
Both doctor and admin have access to it. So I don't think it should be adminAuthenticator.

Reply:
But doctor will not be part of this HIU at all dude.

Reply:
But this is the same HIU code which will be used for both purposes, right?